templatesProvider.js

```diff
@@ -13,79 +13,105 @@
+const { promisify } = require('util');
+const exec = promisify(require('child_process').exec);
+
+function canHandle(repository) {
+    return repository.id && 
+        Array.isArray(repository.projectStyles) &&
+        repository.projectStyles.includes('Appsody');
+}
 
 module.exports = {
 
-    getRepositories: async function() {
-        return new Promise((resolve, reject) => {
-            
-            exec(`${__dirname}/appsody repo list -o json`, (err, stdout) => {
+    addRepository: async function(repository) {
+        
+        // no-op
+        if (!canHandle(repository) || 
+            !repository.url.endsWith('index.json'))
+            return;
+
+        let url = repository.url;
+        url = url.substring(0, url.length - 10) + 'index.yaml';
+
+        await exec(`${__dirname}/appsody repo add ${repository.id} ${url}`);
+    },
 
-                if (err)
-                    return reject(err);
+    removeRepository: async function(repository) {
+        
+        // no-op
+        if (!canHandle(repository))
+            return;
 
-                const repos = [];
-                const json = JSON.parse(stdout);
+        await exec(`${__dirname}/appsody repo remove ${repository.id}`);
+    },
 
-                for (const repo of json.repositories) {
+    getRepositories: async function() {
+        
+        const repos = [];
+            
+        const result = await exec(`${__dirname}/appsody repo list -o json`);
+        const json = JSON.parse(result.stdout);
 
-                    const name = repo.name;
-                    let url = repo.url;
+        for (const repo of json.repositories) {
 
-                    if (name != 'experimental' && url.endsWith('index.yaml')) {
+            const name = repo.name;
+            let url = repo.url;
 
-                        url = url.substring(0, url.length - 10) + 'index.json';
+            if (name != 'experimental' && url.endsWith('index.yaml')) {
 
-                        repos.push({
-                            name: `Appsody Stacks - ${name}`,
-                            description: 'Use Appsody in Codewind to develop applications with sharable technology stacks.',
-                            url
-                        });
-                    }
-                }
+                url = url.substring(0, url.length - 10) + 'index.json';
 
-                resolve(repos);
-            });
-        });
+                repos.push({
+                    id: name,
+                    name: `Appsody Stacks - ${name}`,
+                    description: 'Use Appsody in Codewind to develop applications with sharable technology stacks.',
+                    url
+                });
+            }
+        }
+
+        return repos;
     },
 
-    getProjectTypes: async function() {
-        return new Promise((resolve, reject) => {
-            
-            exec(`${__dirname}/appsody list -o json`, (err, stdout) => {
+    getProjectTypes: async function(id) {
```

Comment:
Added `id` param here but it cannot be used with `appsody list <id> -o json` until we have an appsody CLI that has fixes for:

https://github.com/appsody/appsody/issues/522
https://github.com/appsody/appsody/issues/523